code/models/models.js

```diff
@@ -89,3 +89,58 @@ exports.updateArticleVotes = (request) => {
     return result.rows;
   });
 };
+
+
+
+
+exports.insertComment = (request) => {
+  const articleID = request.params.articles_id;
+  const commentsReceived = request.body;
+  // construct the comment to insert:
+  let commentToInsert = {
+    //comment_id: --> to be provided by the psql query,
+    body: commentsReceived.body,
+    article_id: articleID,
+    author: commentsReceived.username,
+    votes: 0,
+    // created_at: --> psql will provide by using "DEFAULT NOW()",
+  };
+  let queryString = `
+  INSERT INTO comments
+  (body, author, article_id, votes) 
+  VALUES
+  ('${commentToInsert.body}','${commentToInsert.author}',${commentToInsert.article_id},${commentToInsert.votes})
+  RETURNING *;`;
+  return db.query(queryString).then((result) => {
+    if (result.rowCount === 0) {
+      return Promise.reject("Article not found");
+    }
+    return result.rows;
+  });
+};
+
+exports.selectUsers = () => {
+  let queryString = `
+  SELECT *
+  FROM users`;
+  return db.query(queryString).then((result) => {
+    return result.rows;
+  });
+};
+
+exports.selectOneUser = (req) => {
+  let user = req.body.username;
+  // if (!Number.isInteger(+user)) {
+  //   return Promise.reject("Bad Request - User does not exist");
+  // }
+  let queryString = `
+  SELECT *
+  FROM users
+  WHERE username = '${user}'`;
+  return db.query(queryString).then((result) => {
+    if (result.rowCount === 0) {
+      return Promise.reject("Bad Request - User does not exist");
+    }
+    return result.rows;
+  });
+};
```

Comment:
This can now be deleted once you have refactored number 7. We will only need the one model in the controller for posting.

Reply:
DOne, code refractored.